src/main/scala/fairysf/Situation.scala

```diff
@@ -7,7 +7,7 @@ import cats.implicits._
 
 import strategygames.fairysf.format.Uci
 
-case class Situation(board: Board, player: Player) {
+case class Situation(board: Board, player: Player, lastMove: Option[Move] = None) {
```

Comment:
We should be trying to avoid changing the signature of key classes like Situation wherever possible.
In this case `lastMove` should be something that is tracked within `board.history`. There is `lastTurn` and `currentTurn` and a function that returns `lastAction` (something like [pseudo code] `currentTurn.tailOption otherwise lastTurn.tailOption`)

Reply:
In History, both of these are `Uci` and in this case, it is a `Move`. The reason why it's a move in this cause is because I use `situationBefore` when generating the `Fen` for `Amazons` which allows me to access the board state before this move.  It's not immediately obvious to me how to do that from the `Uci`

Reply:
Oh, nevermind `situationBefore` is on the `game` not the `move`